### ISSUE TITLE
Multiple currencies exchanges and value equations

### ISSUE THREAD

**Issue by numapanuma** [gopacifia/DEEP#1]
How to distribute value equations of different kinds of values (e.g. time + money) to the corresponding virtual accounts..

**Comment by fosterlynn** [comment]
Capturing some history from telegram chat:

Numa
And could we have a multi-currency cost: So workers log their time working on it, inputs values are logged as well, and the final product has a total cost of say 5 hours + 200$
Lynn Foster
the value is interesting, you can't really calculate it without
That depends on what value is based, if based on market price then yeah you need value equation, but if based on costs you don't.
which is what we propose for the time system
Lynn Foster
yes different units will always cause problems - one job of the value equation is to get everything into the same unit
so you will have to get hours into $ or $ into hours
in your example
Numa
but what if agent have different virtual accounts: one in time, one in usd, etc. Could a value equation not convert all into one unit and distribute different units to the corresponding accounts?
Lynn Foster
well the value equation takes something in and splits it up - so if you sell something, and you get money back, it can only split up the money
and same for any unit
i guess if someone pays you in promised hours, you would need to split up the hours
but in any case, all the contributions need to be translated into the same unit, which is the one that you are dividing up
part of the equation does that (there are lots of little equations really)
i guess if you have 2 different units, you could just run 2 value equations
Numa
Lynn Foster
well the value equation takes something in and splits it up - so
but what if payment is done in time + money through the customer's virtual account in time, and virtual account in money? Could it be done technically?
Lynn Foster
i guess if you have 2 different units, you could just run 2 valu
ok :)
Lynn Foster
right now, the distribution code assumes one virtual account per person - (marked as "owner" or something) - and uses that in the distributions - if you want to have more than one virtual account per person with differeent units, we'd just have to make some changes for that
i don't think that would be a really big deal
and for your situation would make sense

Bob Haugen
I think it would work to have multiple value equations, each distributing a different resource: money, hours, whatever.
Might require some minor code changes to put each into a different virtual account for each person. 
But I think that would be easier and mathematically more tractable than trying to get one value equation to split up different resources in the same run.
It is a complex graph traversal plus some math.
.....
Numa
The network we are building is based on the time currency. But when you get a table from the carpenter, the latter has to charge you 4 hours + the money he/she spent to buy materials outside of the network (glue, nails, fuel, etc.). So if we don't have this functionality, we basically can't launch the platform.
Now there's a turn around to modifying value equation:
instead of inputting all contributions to one process and then distribute value equation,
we can work with exchanges:
carpenter (individual) exchange his/her time to the carpenter coop (context agent)
carpenter coop logs money based contributions (resources bought) and then exchange the final product with the customer where the exchange has a money reciprocal transfer and a time reciprocal transfer.
With this trick we can solve it for now :)

**Comment by bhaugen** [comment]
I think if you are basically using a time currency + other currencies as needed, exchanges will work better than value equations. 

The idea for value equations came from Sensorica, where they were making products for the capitalist market, and wanted to avoid having a capitalist management taking a profit off the top. Instead, they wanted the income distribution (which would always be money in their situation) to be decided democratically by the project contributors themselves.

Time currencies tend to be exchanged on-the-spot, from provider to recipient, at the time of receiving whatever is being exchanged.

**Comment by numapanuma** [comment]
That's right Bob, it makes sense. We can focus on exchanges then for now. And we'll see if multiple currencies are needed later on (not as an MVP feature).

**Comment by TiberiusB** [comment]
Very interesting case, and I think it is a real case. 

References: [Access to benefits equation](http://valuenetwork.referata.com/wiki/Access_to_benefits_equation) - is the more generic concept; [value equation](http://valuenetwork.referata.com/wiki/The_Value_Equation)

The value equation implemented by SENSORICA has been designed for a single type of benefit, and the use case is monetary currency, i.e. $. It takes in multiple types of contributions, puts them on the same unit/tokens, and spits out a % for all contributors. The %-ges are used to redistribute the benefit. 

From here, we can think of more complex benefit redistribution equations. 

Single benefit redistribution, but allowing different types of benefits. 
Multiple monetary currencies: 
- Suppose that the coop sales something on the market and accepts $ and bitcoin. The NRP would need different virtual accounts for different currencies. 
Multiple types of benefits: 
- Suppose now that the coop provides a carpenter services to a food store, and the store offers 100$ + 100 bananas. In this case, bananas are perishable, so no need to have a virtual account for bananas, they are distributed and consumes immediately.  Unless the community wants to keep track of the number of bananas that every contributor has received over time. This is similar to a case where $ + time access is received back, as in the case where the coop offers services to a coworking space and it pays back $ and free use time for the coworking space for the coop members. 

I think that the simplest way to distribute multiple types of benefits is to run the value equation multiple times, for every type of benefit, or to design different value equations for each type of benefit. 
At the value equation level, you can do a lot of things. For example, you can allocate a % of a benefit for the coop in a bucket, another % for the individual who found the contract, etc., and distribute the rest to contributors.
